### PR TITLE
feat(rome_js_analyzer): no duplicate jsx props rule

### PR DESCRIPTION
**Pull request by iwollmann**
## Summary

New lint rule to disallow duplicate properties in JSX

Closes https://github.com/rome/tools/issues/4029

## Test Plan

`cargo test -p rome_js_analyze -- no_duplicate_jsx_props`

You can see the results in the valid and invalid snapshots
## Documentation

- [ ] The PR requires documentation
- [ ] I will create a new PR to update the documentation

### CODE REVIEW

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+impl Rule for NoDuplicateJsxProps {
+    type Query = Ast<AnyJsxElement>;
+    type State = (String, Vec<JsxAttribute>);
+    type Signals = HashMap<String, Vec<JsxAttribute>>;
+    type Options = ();
+
+    fn run(ctx: &rome_analyze::context::RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let mut defined_properties: HashMap<String, Vec<JsxAttribute>> = HashMap::new();
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:53]: How about normalizing nomenclature? In some places is `properties`, in other `attributes`... Given that the AST call this attribute, I would name everything attribute.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        for attribute in node.attributes() {
+            match attribute {
+                AnyJsxAttribute::JsxAttribute(attribute) => {
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:56]: I think you can simplify this to

```rust
fn push_attribute(att: JsxAttribute, attributes: HashMap) -> Option<()> {
    let name = attr.name().ok()?.syntax().text_trimmed();
    let name = name.to_string().to_lowercase();
    attributes.entry(name)
        .or_default()
        .push(attribute);
}

for att in node.attributes() {
    if let AnyJsxAttribute::JsxAttribute(att) = att {
        let _ = push_attribute(att, &mut defined_properties);
    }
}
```

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                            .or_default()
+                            .push(attribute);
+                    }
+                }
+                AnyJsxAttribute::JsxSpreadAttribute(_) => continue,
+            }
+        }
+
+        defined_properties.retain(|_, val| val.len() > 1);
+
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:69]: Nit: can kill this empty line.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        defined_properties
+    }
+
+    fn diagnostic(
+        _: &rome_analyze::context::RuleContext<Self>,
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:74]: You import RuleContext and remove the `rome_analyze::context::`

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 5]
```diff
@@ -0,0 +1,94 @@
+use std::collections::HashMap;
+
+use rome_analyze::{declare_rule, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_js_syntax::jsx_ext::AnyJsxElement;
+use rome_js_syntax::{AnyJsxAttribute, JsxAttribute, SyntaxNodeText};
+use rome_rowan::AstNode;
+
+declare_rule! {
+    /// Prevents duplicate properties in JSX elements.
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:10]: Maybe more sense to say:

```
Prevents JSX properties to be assigned multiple times.
```

@ematipico You are better than me in this. :)

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    fn diagnostic(
+        _: &rome_analyze::context::RuleContext<Self>,
+        state: &Self::State,
+    ) -> Option<rome_analyze::RuleDiagnostic> {
+        let mut attributes = state.1.iter();
+
+        let mut diagnostic = RuleDiagnostic::new(
+            rule_category!(),
+            attributes.next()?.syntax().text_trimmed_range(),
+            markup!("Elements can not have attributes with the same name."),
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:82]: How about? 

```
This JSX property is assigned multiple times.
```

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        );
+
+        for attr in attributes {
+            diagnostic = diagnostic.detail(
+                attr.syntax().text_trimmed_range(),
+                "attribute is duplicated!",
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:88]: This attribute is assigned again here.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs [hunk 8]
```diff
@@ -0,0 +1,94 @@
+use std::collections::HashMap;
+
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:2]: Call kill this line.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 1 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs]: Done.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 2 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs]: Done.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 4 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs]: Done.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 8 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs]: Done.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 3 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs]: Done.

File: crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs (hunk 5 above)
Comment (iwollmann) [reply crates/rome_js_analyze/src/analyzers/nursery/no_duplicate_jsx_props.rs:10]: I will wait until @ematipico reviews to make all text adjustments.

**Pull request merged**